cursive/examples/filters.rs

```diff
@@ -0,0 +1,135 @@
+                                        button.fixed_width(10)
+                                    })
+                                    .child({
+                                        let mut button = color_group
+                                            .button(Some(Color::Blue), "Blue");
+                                        if let Some(Color::Blue) =
+                                            current_color
+                                        {
+                                            button.select();
+                                        }
+                                        button.fixed_width(10)
+                                    }),
+                            )
+                            .child(
+                                LinearLayout::horizontal()
+                                    .child(
+                                        TextView::new("Size:")
+                                            .fixed_width(15)
+                                            .fixed_height(2),
+                                    )
+                                    .child(
+                                        size_group
+                                            .button(None, "Any")
+                                            .fixed_width(10),
+                                    )
+                                    .child({
+                                        let mut button = size_group.button(
+                                            Some(Size::Small),
+                                            "Small",
+                                        );
+                                        if let Some(Size::Small) = current_size
+                                        {
+                                            button.select();
+                                        }
+                                        button.fixed_width(10)
+                                    })
+                                    .child({
+                                        let mut button = size_group.button(
+                                            Some(Size::Large),
+                                            "Large",
+                                        );
+                                        if let Some(Size::Large) = current_size
+                                        {
+                                            button.select();
+                                        }
+                                        button.fixed_width(10)
+                                    }),
+                            ),
+                    )
+                    .button("Done", move |s| {
+                        // Save selected filters as user_data
+                        s.user_data::<Filters>().unwrap().color =
+                            *color_group.selection();
+                        s.user_data::<Filters>().unwrap().size =
+                            *size_group.selection();
```

Comment:
Here too, `with_user_data` may help:

```suggestion
                        s.with_user_data(|filters: &mut Filters| {
                            filters.color = *color_group.selection();
                            filters.size = *size_group.selection();
                        }).unwrap(); 
```